### PR TITLE
docs: Dedicated guide for configuring the Helm chart

### LINKED ISSUE #1194: Provide more details on how to customize Helm charts
### Describe the Issue

Some users may not be familiar with Helm or how they can/should configure the charts.

We should provide more context in any relevant topics by taking some details from this Helm doc: https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing

### Updates to existing documentation

Any guides that deal with deploying Redpanda using Helm.

We should also put all config related topics inside a parent that explains how to configure the chart so that users can find it through the breadcrumbs.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: February 25

Fixes #1194

### CODE REVIEW

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+```yaml title="console-enable-admin-api.yaml"
+console:
+  enabled: true
+  config: 
+    console:
+      redpanda:
+        adminApi:
+          enabled: true
+          urls:
+          - http://redpanda-0.redpanda.redpanda.svc.cluster.local.:9644
```
Comment (RafalKorepta) [inline docs/manage/kubernetes/configure-helm-chart.mdx:104]: This is good example, but @joejulian shouldn't this be default for console integration in redpanda chart?

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+For more details, see the [Helm documentation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).
+
+The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.
+
+</TabItem>
+</Tabs>
+
+## Configure Redpanda Console
+
+You can configure Redpanda Console in the `console.config` object of the Redpanda Helm chart using the usual [configuration values](../../../reference/console/config).
```
Comment (RafalKorepta) [inline docs/manage/kubernetes/configure-helm-chart.mdx:88]: I wonder if `[configuration values](../../../reference/console/config)` is enough to mention here. Or we could add 
```console
helm repo add redpanda https://charts.redpanda.com
helm repo update
helm show values redpanda/console
```

Just to mention it, but encourage users to use the integration between redpanda and console chart.

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+- Your overrides
+- The values in the existing release
+- The default values in the new release (if you're upgrading with `helm upgrade`)
+
+To see what options you can override in the chart, use the `helm show values` command:
+
+```bash
+helm repo add redpanda https://charts.redpanda.com
+helm repo update
+helm show values redpanda/redpanda
```
Comment (RafalKorepta) [inline docs/manage/kubernetes/configure-helm-chart.mdx:31]: Should we mention visual way/browser view of our values: https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values

File: docs/manage/kubernetes/configure-helm-chart.mdx (hunk 2 above)
Comment (RafalKorepta) [reply docs/manage/kubernetes/configure-helm-chart.mdx:88]: Should we mention visual way/browser view of our values: https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values

File: docs/manage/kubernetes/configure-helm-chart.mdx (hunk 1 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/configure-helm-chart.mdx:104]: Joe said that he's going to make this the default. I can update this example when it's done.

File: docs/manage/kubernetes/configure-helm-chart.mdx (hunk 3 above)
Comment (JakeSCahill) [reply docs/manage/kubernetes/configure-helm-chart.mdx:31]: The `<ValuesYaml>` component below this code injects this link.

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+<ValuesYaml/>
+
+:::
+
+<Tabs>
+<TabItem value="values" label="--values" default>
+
+The `--values` option allows you to keep your overrides in one or more YAML files. 
+If you provide multiple files to this option and you override the same values in two or more of them, the rightmost file takes precedence.
+For example, you might override the `storage.persistentVolume.storageClass` configuration in a file called `storage-class.yaml`:
```
Comment (Feediver1) [inline docs/manage/kubernetes/configure-helm-chart.mdx:45]:
```suggestion
The `--values` option enables you to keep your overrides in one or more YAML files. 
If you specify multiple files to this option and then override the same values in two or more of them, the rightmost file takes precedence.
For example, you might override the `storage.persistentVolume.storageClass` configuration in a file called `storage-class.yaml`:
```

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+The `helm` command to apply this configuration override looks something like the following:
+
+```bash
+helm upgrade --install redpanda redpanda/redpanda \
+    --namespace redpanda --create-namespace \
+    --values storage-class.yaml --reuse-values
+```
+
+The values in your YAML files override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.
```
Comment (Feediver1) [inline docs/manage/kubernetes/configure-helm-chart.mdx:61]:
```suggestion
The values in your YAML files override their counterparts in the Helm chart’s `values.yaml` file. Any values not overridden maintain their defaults.
```

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+Use the `--reuse-values` flag to apply your overrides on top of existing overrides that you've already made. Don't include this flag if you're upgrading to a new version of the Helm chart. If you're upgrading to a new version of the Helm chart, this flag prevents any values in the new release from being applied.
+
+If you're upgrading and you already have Redpanda Console installed, set `console.enabled` to `false` to stop Helm from trying to deploy it again.
+
+</TabItem>
+<TabItem value="set" label="--set" default>
+
+The `--set` option allows you to specify configuration overrides in the command line.
```
Comment (Feediver1) [inline docs/manage/kubernetes/configure-helm-chart.mdx:70]:
```suggestion
The `--set` option enables you to specify configuration overrides in the command line. 
```

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+```bash
+helm upgrade --install redpanda redpanda/redpanda \
+    --namespace redpanda --create-namespace \
+    --set storage.persistentVolume.storageClass=my-storage-class
+```
+
+For more details, see the [Helm documentation](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).
+
+The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values that weren’t overridden keep their defaults.
```
Comment (Feediver1) [inline docs/manage/kubernetes/configure-helm-chart.mdx:81]:
```suggestion
The values in the `--set` options override their counterparts in the Helm chart’s `values.yaml` file. Any values not overridden maintain their defaults.
```

File: docs/manage/kubernetes/configure-helm-chart.mdx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+If you want to use the separate Redpanda Console Helm chart, disable Redpanda Console in the Redpanda Helm chart with `console.enabled=false`.
+To see what options you can override in the Redpanda Console chart, use the `helm show values` command:
+
+```bash
+helm repo add redpanda https://charts.redpanda.com
+helm repo update
+helm show values redpanda/console
+```
+
+For default values and documentation for configuration options, see the [\`values.yaml\`](https://artifacthub.io/packages/helm/redpanda-data/console?modal=values) file.
```
Comment (Feediver1) [inline docs/manage/kubernetes/configure-helm-chart.mdx:137]: Are you intending for the format tags to appear around values.yaml here? Just checking.

File: versioned_docs/version-22.3/manage/index.mdx [hunk 9]
```diff
@@ -9,6 +9,10 @@ title: Manage
 
 - [Kubernetes](../manage/kubernetes)
 
+- [Customize the Helm Chart](../manage/kubernetes/configure-helm-chart)
+
+    You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys.
```
Comment (Feediver1) [inline versioned_docs/version-22.3/manage/index.mdx:14]:
```suggestion
    You can customize the Redpanda Helm chart to configure which cluster and the Kubernetes components the chart deploys.
```

File: versioned_docs/version-22.3/manage/kubernetes/configure-helm-chart.mdx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+<head>
+    <meta name="title" content="Customize the Helm Chart | Redpanda Docs"/>
+    <meta name="description" content="You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys."/>
+</head>
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ValuesYaml from './shared/_values-yaml.mdx'
+
+You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys.
```
Comment (Feediver1) [inline versioned_docs/version-22.3/manage/kubernetes/configure-helm-chart.mdx:16]:
```suggestion
You can customize the Redpanda Helm chart to configure which cluster and Kubernetes components the chart deploys.
```

File: versioned_docs/version-22.3/manage/kubernetes/index.mdx [hunk 11]
```diff
@@ -7,6 +7,10 @@ title: Manage Redpanda Self-Hosted in Kubernetes
     <meta name="description" content="Manage Self-Hosted Redpanda in Kubernetes."/>
 </head>
 
+- [Customize the Helm Chart](../kubernetes/configure-helm-chart)
+
+    You can customize the Redpanda Helm chart to configure the cluster and the Kubernetes components that the chart deploys.
```
Comment (Feediver1) [inline versioned_docs/version-22.3/manage/kubernetes/index.mdx:12]:
```suggestion
    You can customize the Redpanda Helm chart to configure which cluster and Kubernetes components the chart deploys.
```

**Review by Feediver1 — APPROVED** [review]
suggestions apply to both versions. Your call to implement or not.

File: docs/manage/kubernetes/configure-helm-chart.mdx (hunk 8 above)
Comment (JakeSCahill) [inline docs/manage/kubernetes/configure-helm-chart.mdx:137]:
```suggestion
For default values and documentation for configuration options, see the [`values.yaml`](https://artifacthub.io/packages/helm/redpanda-data/console?modal=values) file.
```

**Pull request merged**